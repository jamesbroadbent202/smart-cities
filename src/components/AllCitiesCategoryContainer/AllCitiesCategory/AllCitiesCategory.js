import React from 'react';
import PropTypes from 'prop-types';
import flatten from 'lodash/flatten';
import PageWrapper from '../../PageWrapper/PageWrapper';
import PageBanner from '../../PageBanner/PageBanner';
import SubCategorySummary from '../../SubCategorySummary/SubCategorySummary';
import SubCategoryDetails from '../../SubCategoryDetails/SubCategoryDetails';
import CityColumnChart from '../../CityColumnChart/CityColumnChart';
import { INDICATORS } from '../../../constants';
import style from './AllCitiesCategory.scss';

const AllCitiesCategory = (props) => {
  const isHeroInSubCategory = props.category.subCategories.find(sc => (
    flatten(sc.charts.map(chart => chart.indicatorIds)).includes(props.category.heroIndicatorId)
  )) !== undefined;

  return (
    <PageWrapper categoryId={props.category.id}>
      <PageBanner
        colorName={props.category.colorName}
        description={props.category.description}
        indicator={props.category.heroIndicatorId}
        title={props.category.name}
        cities={props.cities}
        isCategoryPage
      />

      {props.category.subCategories.map(subCategory => (
        <SubCategorySummary
          key={subCategory.name.replace(' ', '-')}
          {...subCategory}
          colorName={props.category.colorName}
          cities={props.cities}
        />
      ))}

      {isHeroInSubCategory || <div className={style.heroWrapper}>
        <CityColumnChart
          cities={props.cities}
          colorBase={props.category.colorName}
          colorVariation={500}
          indicatorIds={[props.category.heroIndicatorId]}
        />
      </div>}

      {props.category.subCategories.map(subCategory => (
        <SubCategoryDetails
          key={subCategory.name}
          subCategory={subCategory}
          colorName={props.category.colorName}
          cities={props.cities}
          heroIndicatorId={props.category.heroIndicatorId}
        />
      ))}
    </PageWrapper>
  );
};

AllCitiesCategory.propTypes = {
  category: PropTypes.shape({
    colorName: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    heroIndicatorId: PropTypes.oneOf(Object.keys(INDICATORS)).isRequired,
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    subCategories: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      charts: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string.isRequired,
        indicatorIds: PropTypes.arrayOf(
          PropTypes.oneOf(Object.keys(INDICATORS)),
        ).isRequired,
      })),
    })).isRequired,
  }).isRequired,
  cities: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    indices: PropTypes.object.isRequired,
  })).isRequired,
};

export default AllCitiesCategory;
