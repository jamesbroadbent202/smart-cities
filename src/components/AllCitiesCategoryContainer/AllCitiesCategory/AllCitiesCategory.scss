@import '~style/variables';

.heroWrapper {
  background: $color-white;
}
