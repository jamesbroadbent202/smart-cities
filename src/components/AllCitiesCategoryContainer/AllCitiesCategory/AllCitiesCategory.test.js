import React from 'react';
import { shallow } from 'enzyme';
import AllCitiesCategory from './AllCitiesCategory';

jest.mock('../../PageBanner/PageBanner', () => 'PageBanner');
jest.mock('../../SubCategorySummary/SubCategorySummary', () => 'SubCategorySummary');

const defaultProps = {
  category: {
    colorName: 'jobs',
    description: 'This is the mock category',
    heroIndicatorId: 'population',
    id: 'mock',
    name: 'Mock category',
    subCategories: [
      {
        name: 'Buying houses',
        tint: '100',
        summaryIndicatorIds: [],
        charts: [],
      },
      {
        name: 'Living in houses',
        tint: '050',
        summaryIndicatorIds: [],
        charts: [],
      },
    ],
  },
  cities: [
    {
      name: 'Perth',
      indices: {}
    },
    {
      name: 'Sydney',
      indices: {}
    },
  ],
};

it('should match Snapshot', () => {
  const component = shallow(
    <AllCitiesCategory {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});
