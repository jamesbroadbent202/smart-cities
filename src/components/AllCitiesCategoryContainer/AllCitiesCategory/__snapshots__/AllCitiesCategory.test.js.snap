// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<PageWrapper categoryId=\\"mock\\" cityId=\\"all-cities\\">
  <PageBanner colorName=\\"jobs\\" description=\\"This is the mock category\\" indicator=\\"population\\" title=\\"Mock category\\" cities={{...}} isCategoryPage={true} isContextPage={false} />
  <SubCategorySummary name=\\"Buying houses\\" highlightColorDark=\\"JOBS_600\\" summaryIndicatorIds={{...}} charts={{...}} categoryId=\\"mock\\" cities={{...}} />
  <SubCategorySummary name=\\"Living in houses\\" highlightColorDark=\\"JOBS_700\\" summaryIndicatorIds={{...}} charts={{...}} categoryId=\\"mock\\" cities={{...}} />
  <div className=\\"heroWrapper\\">
    <CityColumnChart cities={{...}} colorBase=\\"jobs\\" highlightColorDark=\\"MOCK_500\\" indicatorIds={{...}} />
  </div>
  <SubCategoryDetails subCategory={{...}} colorName=\\"jobs\\" cities={{...}} heroIndicatorId=\\"population\\" highlightColorDark=\\"JOBS_600\\" />
  <SubCategoryDetails subCategory={{...}} colorName=\\"jobs\\" cities={{...}} heroIndicatorId=\\"population\\" highlightColorDark=\\"JOBS_700\\" />
</PageWrapper>"
`;
